libs/dh/admin/feature-user-management-new/src/lib/details/tabs/audit-logs.component.ts

```diff
@@ -48,54 +44,39 @@ import { DhUser } from '@energinet-datahub/dh/admin/shared';
       .no-results-text {
         text-align: center;
       }
-
-      .spinner {
-        display: flex;
-        justify-content: center;
-      }
     `,
   ],
   imports: [
-    RxPush,
     TranslocoPipe,
-    NgTemplateOutlet,
     TranslocoDirective,
 
     WATT_CARD,
     WATT_TABLE,
     WattDatePipe,
-    WattSpinnerComponent,
     WattEmptyStateComponent,
   ],
 })
-export class DhUserAuditLogsComponent implements OnChanges {
-  private readonly apollo = inject(Apollo);
-  private readonly getUserAuditLogsQuery = this.apollo.watchQuery({
-    query: GetUserAuditLogsDocument,
-    fetchPolicy: 'cache-and-network',
-  });
-
-  readonly dataSource = new WattTableDataSource<UserAuditedChangeAuditLogDto>();
+export class DhUserAuditLogsComponent {
+  private getUserAuditLogsQuery = lazyQuery(GetUserAuditLogsDocument);
 
-  hasFailed$ = this.getUserAuditLogsQuery.valueChanges.pipe(
-    map((result) => !!result.error),
-    catchError(() => of(true))
-  );
+  user = input.required<DhUser>();
 
-  isLoading$ = this.getUserAuditLogsQuery.valueChanges.pipe(
-    tap((result) => (this.dataSource.data = [...(result.data?.userAuditLogs ?? [])].reverse())),
-    map((result) => result.loading),
-    catchError(() => of(false))
-  );
+  dataSource = new WattTableDataSource<UserAuditedChangeAuditLogDto>();
+  hasError = computed(() => this.getUserAuditLogsQuery.error() !== undefined);
+  isLoading = this.getUserAuditLogsQuery.loading;
 
   columns: WattTableColumnDef<UserAuditedChangeAuditLogDto> = {
     timestamp: { accessor: 'timestamp' },
     entry: { accessor: null },
   };
 
-  user = input.required<DhUser>();
+  constructor() {
+    effect(() => {
+      this.getUserAuditLogsQuery.query({ variables: { id: this.user().id } });
+    });
 
-  ngOnChanges(): void {
-    this.getUserAuditLogsQuery?.refetch({ id: this.user().id });
+    effect(() => {
+      this.dataSource.data = this.getUserAuditLogsQuery.data()?.userAuditLogs || [];
```

Comment:
issue: in the old version the audit log entries were reversed. Let's keep it.

Reply:
Fixed